lib/qtrajectory.h

```diff
@@ -279,50 +275,38 @@ class QuantumTrajectorySimulator {
 
       NormalizeState(!unitary, state_space, unitary, state);
 
-      if (state_space.IsNull(scratch) || scratch.num_qubits() < num_qubits) {
-        scratch = CreateState(num_qubits, state_space);
-        if (state_space.IsNull(scratch)) {
-          return false;
-        }
-      }
-
-      state_space.Copy(state, scratch);
+      double max_prob = 0;
+      std::size_t max_prob_index = 0;
 
       // Perform sampling of Kraus operators using norms of updated states.
       for (std::size_t i = 0; i < channel.size(); ++i) {
         const auto& kop = channel[i];
 
         if (kop.unitary) continue;
 
-        // Apply the Kraus operator.
-        if (kop.ops.size() == 1) {
-          ApplyGate(simulator, kop.ops[0], state);
-        } else {
-          DeferOps(kop.ops, gates);
+        double prob = std::real(
+            simulator.ExpectationValue(kop.qubits, kop.kd_k.data(), state));
```

Comment:
[No change needed] This is really clever! If I understand correctly, this works because:
```
prob(Ψ, K) = abs(K|Ψ>) ** 2 = <Ψ|K†K|Ψ> = EV(Ψ, K†K)
```
Is that correct?

Reply:
Yes, that is correct.